### PR TITLE
options/posix: implement pthread_mutexattr

### PR DESCRIPTION
**Pull request by 64**
Progress on #530.

### CODE REVIEW

File: options/posix/generic/pthread-stubs.cpp [hunk 1]
```diff
@@ -688,67 +688,95 @@ int pthread_once(pthread_once_t *once, void (*function) (void)) {
...
+	attr->__mlibc_pshared = PTHREAD_PROCESS_PRIVATE;
+	attr->__mlibc_protocol = PTHREAD_PRIO_NONE;
 	return 0;
 }
 
-int pthread_mutexattr_destroy(pthread_mutexattr_t *) {
+int pthread_mutexattr_destroy(pthread_mutexattr_t *attr) {
 	SCOPE_TRACE();
-
+	memset(attr, 0, sizeof *attr);
```
Comment (ArsenArsen) [inline options/posix/generic/pthread-stubs.cpp:700]: we use function style sizeof

**Review by ArsenArsen — CHANGES_REQUESTED** [review]

File: options/posix/generic/pthread-stubs.cpp (hunk 1 above)
Comment (64) [reply options/posix/generic/pthread-stubs.cpp]: not at all consistently :^)

**Review by Geertiebear — APPROVED** [review]
LGTM.

**Comment by Geertiebear** [comment]
Does this complete the mutexattr_t struct?

**Comment by 64** [comment]
> Does this complete the mutexattr_t struct?

We have
```c
struct __mlibc_mutex {
	unsigned int __mlibc_state;
	unsigned int __mlibc_recursion;
	unsigned int __mlibc_flags;
};
typedef struct __mlibc_mutex pthread_mutex_t;

struct  __mlibc_mutexattr {
	int __mlibc_type;
	int __mlibc_robust;
	int __mlibc_protocol;
	int __mlibc_pshared;
	int __mlibc_prioceiling;
};
typedef struct __mlibc_mutexattr pthread_mutexattr_t;
```

I guess it's just a question of which fields we can compress into `__mlibc_flags` here (I think all of them except `prioceiling`?)

**Comment by 64** [comment]
Oh disregard, yes `mutexattr_t` should be complete now but `mutex_t` is not necessarily (as per the above).

**Comment by Geertiebear** [comment]
Then let's add a required field for prioceiling too, to avoid an ABI break in the future.

**Comment by 64** [comment]
> Then let's add a required field for prioceiling too, to avoid an ABI break in the future.

Done. I've also made the getprioceiling / setprioceiling functions return EINVAL (matching musl) since we don't implement that behaviour and it appears to be a pretty heavyweight feature.

**Review by Geertiebear — APPROVED** [review]
LGTM now then.

**Pull request merged**